comment/mixins.py

```diff
@@ -1,162 +1,143 @@
 import abc
 
-from django.contrib.auth.mixins import LoginRequiredMixin, AccessMixin
 from django.core.exceptions import ImproperlyConfigured
-from django.http import HttpResponseBadRequest, HttpResponseForbidden
-from django.views.generic import FormView
 
 from comment.conf import settings
-from comment.utils import is_comment_admin, is_comment_moderator
+from comment.utils import is_comment_admin, is_comment_moderator, can_block_user, can_moderate_flagging
 from comment.validators import ValidatorMixin
-from comment.messages import ErrorMessage, FlagError, FollowError
-from comment.service.email import DABEmailService
-from comment.messages import EmailInfo
-from comment.forms import CommentForm
-from comment.context import DABContext
-from comment.responses import DABResponseData
+from comment.messages import ErrorMessage, FlagError, FollowError, BlockedUserMSG
+from comment.responses import UTF8JsonResponse
+from comment.models import BlockedUser
 
 
 class AJAXRequiredMixin:
+    status = 403
```

Comment:
if AJAX request is not allowed, should this be a bad request(400) or forbideen(403)?

Reply:
I believe it should be forbidden, 400 most likely for invalid data.

Reply:
also, keep in mind, this sort of change will make it backwards incompatible, because earlier, we used to return 400.

Reply:
In fact the error were not handled in the JS earlier, it was kept to be caught in the raised error in JS function. Previously we used `HttpResponseBadRequest` which is a text response while we are parsing the response as json in the JS functions.